specs/corla.pvs

```diff
@@ -0,0 +1,923 @@
+%%% order is induced upon them by a random shuffle.
+  ballot_order: TYPE = list[ballot]
+
+%%% Sometimes, but not always, a total order is induced by ballot ids.
+% @todo Introduce something like
+% < (b1: ballot, b2: ballot): bool
+% ballot_ids_unique_and_induce_order: POSTULATE strict_total_order?(<)
+
+END ballots_collections
+
+%+END ballots
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%% Elections have all kinds of different published instructions,
+%%% forms, and reports. We enumerate a few here that are specific to
+%%% Colorado, and the RLA Tool in particular.
+
+instructions_forms_reports: THEORY
+BEGIN
+  audit_report: TYPE
+  ballot_instructions: TYPE
+  sos_audit_form: TYPE
+  sos_voter_intent_guide: TYPE
+END instructions_forms_reports
+
+%%% Within the RLA Tool there are a number of roles that a person
+%%% involved with the system can take on. Some of these roles are
+%%% mutually exclusive due to legal reasons.
+
+roles: THEORY
+BEGIN
+  IMPORTING elections
+
+  audit_board_member: TYPE FROM person
+  canvas_board: TYPE = set[person]
+
+  colorado_department_of_state: TYPE
+  cdos: TYPE = colorado_department_of_state
+
+  sos: TYPE FROM person
+  state?: [sos -> state]
+
+  county_clerk: TYPE FROM person
+  county?: [county_clerk -> county]
+
+% @review CDOS Roles that are currently under-defined and need
+% clarity. In particular, are there any lurking restrictions in roles
+% such that a person cannot have two kinds of roles
+% simultaneously. E.g., can a candidate be an audit board of their own
+% election? Can an county administrator be a candidate? Can the SOS be
+% an audit board member? Etc.
+  administrator: TYPE FROM person
+  audit_supervisor: TYPE FROM person
+% @contraint CDOS states that they want to start with at least one
+% designated official per county, but must have the ability to create
+% credentials for additional users.
+% @trace email from Jerome Lovato on 12 July 2017
+  county_administrator: TYPE FROM person
+  state_administrator: TYPE FROM person
+  
+% @review CDOS Who bootstraps authentication? Who has the right to
+% identify people who should have credentials and generate and issue
+% those credentials? How do they (singular or plural) authenticate to
+% the system. We'll put in place a bootstrapping role for the moment,
+% the system administrator.
+  system_administrator: TYPE FROM person
+END roles
+
+canvas_boards: THEORY
+BEGIN
+  IMPORTING roles, election
+
+  canvas_board?: [election -> canvas_board]
+END canvas_boards
+
+randomness: THEORY
+BEGIN
+  unit: TYPE
+  seed: TYPE = nat
+  rng: TYPE = [unit -> nat]
+  seeded_prng: TYPE = [seed -> rng]
+END randomness
+
+cryptography: THEORY
+BEGIN
+  hash: TYPE
+END cryptography
+
+audits: THEORY
+BEGIN
+  IMPORTING roles
+  
+  audit: TYPE
+%%% @review CDOS Must audit boards have a minimum number of members?  A maximum?
+  audit_board: TYPE = set[audit_board_member]
+  audit_mistake: TYPE
```

Comment:
I don't know what it is yet.  It was used as a phrase in one of the documents I was reading from CDOS.